app/models/concerns/public_s3.rb

```diff
@@ -60,16 +60,16 @@ def #{name}=(attachable)
         # When building a blob we must ensure the correct S3 service is being specified so that the backing code
         # uploads to the correct bucket. Otherwise it will use the default configured in the config/storage.yml
         def upload_public_#{name}(file)
-          blob = ActiveStorage::Blob.new.tap do |blob|
```

Comment:
Quite a few Active Storage changes in 6.1, so I had to redo how we set up blobs. Namely, the service cannot be set directly and is immutable, so you construct it at build time.